src/common/modal/index.tsx

```diff
@@ -86,6 +87,7 @@ const Modal: React.FC<ModalProps> = ({
   showCloseButton,
   actions,
   children,
+  Component,
```

Comment:
I'm 100% not sold on having a "Component" prop.

2 Alternatives here:
- We can change the actions definition to accept a `split` property where that defines wether we show a split button or a normal button.
- We change the actions definition to either accept the object it accepts right now, Buttons and SplitButtons, this would end up having a type of
```
interface ActionsObject {
    label: React.ReactNode;
    onClick: () => void;
    disabled?: boolean;
    color?: keyof typeof ButtonTypes;
    variant?: 'text' | 'outlined' | 'contained';
  }[];
  
  type actions = (ActionsObject | Button | SplitButton)[]
  ```
  
  Then when we render it we just check for the type and render the appropiate child, if its an ActionObject it renders it like we have it right now, and if its a button or SplitButton the map just returns that